### DISCUSSION TITLE
Client IP Address Not Showing in Logs When Using Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by mplawner** [kimai/kimai#5020] [Docker]
### Describe the problem

Hi Kimai Team,

I’m encountering an issue where the Kimai log file only shows the Docker network IP address (`172.20.0.1`) instead of the real client IP address for all requests. This is causing problems with Fail2ban, as it can't correctly identify the source IPs.

### Setup Details

- **Environment:** 
  - Kimai Docker image: `kimai/kimai2:apache`
  - Reverse Proxy: Nginx Proxy Manager with the following headers set:
    ```nginx
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    ```

### Issue Description

Despite configuring Nginx Proxy Manager with the appropriate headers, Kimai's log continues to show the Docker network IP (`172.20.0.1`) as the source for all requests, instead of the actual client IP.

### Troubleshooting Steps Taken

I’ve tried various configurations with the `TRUSTED_PROXIES`, `APP_TRUSTED_PROXIES`, `TRUSTED_HEADERS`, and `APP_TRUSTED_HEADERS` environment variables without success.

The only workaround that has shown the real client IPs in the logs involves manually modifying the Apache configuration within the Docker container:

1. **Modifying `000-default.conf`:**
   I added the following lines to `/etc/apache2/sites-enabled/000-default.conf`:

   ```apache
   # Trust the forwarded headers from the reverse proxy
   SetEnvIf X-Forwarded-Proto https HTTPS=on
   RequestHeader set X-Forwarded-Proto "https"
   RequestHeader set X-Forwarded-For "%{REMOTE_ADDR}s"
   
   RemoteIPHeader X-Forwarded-For
   RemoteIPTrustedProxy 172.20.0.0/16
   ```

2. **Enabling `mod_headers` and `mod_remoteip`:**
   After modifying the config, I entered the Docker container and ran the following command:

   ```bash
   a2enmod headers && a2enmod remoteip && service apache2 restart
   ```

   After this, the real client IPs began showing up in the logs correctly.

### Questions

1. Is this behavior a limitation of the current Kimai Docker image?
2. Am I missing a configuration step, or is there a recommended way to handle this without manual intervention each time the container restarts?

Any guidance or suggestions would be greatly appreciated. If this requires changes to the Docker image, I’d be happy to help test any proposed solutions.

Thank you for your support!

Best regards,  
Michael

### Describe your setup and add your Docker compose file (redact your credentials)

services:
  sqldb:
    image: mysql:8.3
    volumes:
      - ./mysql:/var/lib/mysql
    environment:
      - MYSQL_DATABASE=kimai
      - MYSQL_USER=kimaiuser
      - MYSQL_PASSWORD=${MYSQL_PASSWORD}
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
    command: --default-storage-engine innodb
    restart: unless-stopped
    healthcheck:
      test: mysqladmin -p$$MYSQL_ROOT_PASSWORD ping -h localhost
      interval: 20s
      start_period: 10s
      timeout: 10s
      retries: 3

  kimai:
    image: kimai/kimai2:apache
    ports:
      - 8001:8001
    environment:
      - ADMINMAIL=${ADMINMAIL}
      - ADMINPASS=${ADMINPASS}
      - "DATABASE_URL=mysql://kimaiuser:${MYSQL_PASSWORD}@sqldb:3306/kimai?charset=utf8mb4&serverVersion=8.3.0"
      - TRUSTED_HOSTS=fqdn.hostname.com,localhost,127.0.0.1
      - APP_ENV=prod
      - APP_SECRET=${APP_SECRET}
      - TRUSTED_PROXIES=172.20.0.0/16
      - TRUSTED_HEADERS=x-forwarded-for,x-forwarded-proto
      - APP_TRUSTED_PROXIES=172.20.0.0/16
      - APP_TRUSTED_HEADERS=x-forwarded-for,x-forwarded-proto

    volumes:
      - ./data:/opt/kimai/var/data
      - ./plugins:/opt/kimai/var/plugins
      - ./log:/opt/kimai/var/log
      - ./000-default.conf:/etc/apache2/sites-available/000-default.conf
    restart: unless-stopped

**Comment by kevinpapst** [comment]
> Is this behavior a limitation of the current Kimai Docker image?

Is that a rhetorical question?

> Am I missing a configuration step, or is there a recommended way to handle this without manual intervention each time the container restarts?

> Any guidance or suggestions would be greatly appreciated. If this requires changes to the Docker image, I’d be happy to help test any proposed solutions.

You did the first step by explaining the issue and posting a fix. 
Would be great if you could adapt the Docker image as needed and send in a Pull request, so we can check it.

**Reply by mplawner** [reply]
Hi Kevin,

Thanks for replying.

No, my question wasn't rhetorical, and probably the two questions could have been combined into simply asking: Am I doing something wrong?

However, your reply here and your tagging of the FR suggests that I haven't overlooked something. I'm not very familiar with docker building, but I'll see if I can figure it out.

Thanks!

**Reply by kevinpapst** [reply]
Hey 👋  
It sounded rhetorical, because you ran into an issue and were able to fi it by adapting the image, so to me it looks like "an issue".

I appreciate your long post and explanation, but I am not a Docker user myself, so I'd even appreciate it more if you could share your solution. 

I think you should append your commands here: 
https://github.com/kimai/kimai/blob/main/.docker/Dockerfile.kimai-base#L154
